lightkurve/correctors/pldcorrector.py

```diff
@@ -82,199 +116,509 @@ class PLDCorrector(object):
+        The pixel aperture mask that will be used to create the regression matrix
+        (i.e. the design matrix) used to model the systematics.  If `None`,
+        then the `aperture_mask` value will be used.
+
+    Raises
+    ------
+    ImportError : if one of Lightkurve's optional dependencies required by
+        this class are not available (i.e. `pymc`, `theano`, or `exoplanet`).
     """
-    def __init__(self, tpf):
-        self.tpf = tpf
+    def __init__(self, tpf, aperture_mask=None, pld_aperture_mask=None):
+        # Ensure the optional dependencies requires by this class are installed
+        success, messages = self._check_optional_dependencies()
+        if not success:
+            for message in messages:
+                log.error(message)
+            raise ImportError("\n".join(messages))
+
+        # Input validation: parse the aperture masks to accept strings etc.
+        self.aperture_mask = tpf._parse_aperture_mask(aperture_mask)
+        if pld_aperture_mask is None:
+            self.pld_aperture_mask = self.aperture_mask
+        else:
+            self.pld_aperture_mask = tpf._parse_aperture_mask(pld_aperture_mask)
+        # Generate raw flux light curve from desired pixels
+        raw_lc = tpf.to_lightcurve(aperture_mask=self.aperture_mask)
+        # It is critical to remove all NaNs or the linear algebra below will crash
+        self.raw_lc, self.nan_mask = raw_lc.remove_nans(return_mask=True)
+        self.tpf = tpf[~self.nan_mask]
+        # For user-friendliness, we will track the most recently used model and solution:
+        self.most_recent_model = None
+        self.most_recent_solution_or_trace = None
+
+    def _check_optional_dependencies(self):
+        """Emits a user-friendly error message if one of Lightkurve's
+        optional dependencies which are required for PLDCorrector are missing.
+
+        Returns
+        -------
+        success : bool
+            True if all optional dependencies are available, False otherwise.
+        message : list of str
+            User-friendly error message if success == False.
+        """
+        success, messages = True, []
+
+        try:
+            import pymc3 as pm
+        except ImportError:
+            success = False
+            messages.append("PLDCorrector requires pymc3 to be installed (`pip install pymc3`).")
+
+        try:
+            import exoplanet as xo
+        except ImportError:
+            success = False
+            messages.append("PLDCorrector requires exoplanet to be installed (`pip install exoplanet`).")
 
-        self.flux = tpf.flux
-        self.flux_err = tpf.flux_err
-        self.time = tpf.time
+        try:
+            import theano.tensor as tt
+        except ImportError:
+            success = False
+            messages.append("PLDCorrector requires theano to be installed (`pip install theano`).")
+
+        return success, messages
+
+    def create_first_order_matrix(self):
```

Comment:
Does this need to be a user facing function? Should this be `_create_first_order_matrix`?

Reply:
I'm making this a hidden function, but leaving `create_design_matrix` so users can more easily edit design matrices and append new columns. Calling `create_design_matrix(pld_order=1)` has the same effect as calling this function.